Explorer/Assets/Scripts/CrdtEcsBridge/Serialization/ProtobufSerializer.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Google.Protobuf;
+using System;
+
+namespace CrdtEcsBridge.Serialization
+{
+    public class ProtobufSerializer<T> : IComponentSerializer<T> where T: class, IMessage<T>, new()
+    {
+        public void DeserializeInto(T instance, in ReadOnlySpan<byte> data)
+        {
+            instance.MergeFrom(data);
```

Comment:
I don't think this was ever resolved:

https://github.com/protocolbuffers/protobuf/issues/7885

Reply:
Look at `TextShape.gen.cs` for example:

 ```
void pb::IBufferMessage.InternalMergeFrom(ref pb::ParseContext input) {
      uint tag;
      while ((tag = input.ReadTag()) != 0) {
        switch(tag) {
          default:
            _unknownFields = pb::UnknownFieldSet.MergeFieldFrom(_unknownFields, ref input);
            break;
          case 10: {
            Text = input.ReadString();
            break;
          }
          case 16: {
            Font = (global::DCL.ECSComponents.Font) input.ReadEnum();
            break;
          }
          case 29: {
            FontSize = input.ReadFloat();
            break;
          }
          case 32: {
            FontAutoSize = input.ReadBool();
            break;
          }
          case 40: {
            TextAlign = (global::DCL.ECSComponents.TextAlignMode) input.ReadEnum();
            break;
          }
          case 53: {
            Width = input.ReadFloat();
            break;
          }
          case 61: {
            Height = input.ReadFloat();
            break;
          }
          case 69: {
            PaddingTop = input.ReadFloat();
            break;
          }
          case 77: {
            PaddingRight = input.ReadFloat();
            break;
          }
          case 85: {
            PaddingBottom = input.ReadFloat();
            break;
          }
          case 93: {
            PaddingLeft = input.ReadFloat();
            break;
          }
          case 101: {
            LineSpacing = input.ReadFloat();
            break;
          }
          case 104: {
            LineCount = input.ReadInt32();
            break;
          }
          case 112: {
            TextWrapping = input.ReadBool();
            break;
          }
          case 125: {
            ShadowBlur = input.ReadFloat();
            break;
          }
          case 133: {
            ShadowOffsetX = input.ReadFloat();
            break;
          }
          case 141: {
            ShadowOffsetY = input.ReadFloat();
            break;
          }
          case 149: {
            OutlineWidth = input.ReadFloat();
            break;
          }
          case 154: {
            if (shadowColor_ == null) {
              ShadowColor = new global::Decentraland.Common.Color3();
            }
            input.ReadMessage(ShadowColor);
            break;
          }
          case 162: {
            if (outlineColor_ == null) {
              OutlineColor = new global::Decentraland.Common.Color3();
            }
            input.ReadMessage(OutlineColor);
            break;
          }
          case 170: {
            if (textColor_ == null) {
              TextColor = new global::Decentraland.Common.Color4();
            }
            input.ReadMessage(TextColor);
            break;
          }
        }
      }
    }
```

and `Colors.gen.cs`
```
void pb::IBufferMessage.InternalMergeFrom(ref pb::ParseContext input) {
      uint tag;
      while ((tag = input.ReadTag()) != 0) {
        switch(tag) {
          default:
            _unknownFields = pb::UnknownFieldSet.MergeFieldFrom(_unknownFields, ref input);
            break;
          case 13: {
            R = input.ReadFloat();
            break;
          }
          case 21: {
            G = input.ReadFloat();
            break;
          }
          case 29: {
            B = input.ReadFloat();
            break;
          }
        }
      }
    }
```
Honestly, I am satisfied with this implementation